cancermuts/datasources.py

```diff
@@ -533,12 +533,13 @@ def __init__(self, database_files=None, database_encoding=None):
 
         dataframes = []
 
-        if database_files is None:
-            database_dir   = '/data/databases/cosmic'
-            databases      = [  'CosmicMutantExport' ]
-            self._database_files = [os.path.join(database_dir, i)+'.tsv' for i in databases]
+        if isinstance(database_files, str):
+            self._database_files = [database_files]
         else:
-            self._database_files = database_files
+            if not hasattr(database_files,'__iter__'):
+                self.log.error('COSMIC database_files does not have the correct format.')
+                raise TypeError('COSMIC database_files does not have the correct format.')
+        self._database_files = database_files
```

Comment:
in my testing, this leads to a crash when `database_files` is a string - that's because it's re-assigned here even when it is first assigned after the `if` above (I think you just need to put this statement in the `else` code block)